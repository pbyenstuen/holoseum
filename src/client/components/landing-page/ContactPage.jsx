import React, { useState } from "react";
import InputErrorView from "../shared/InputErrorView";
import InputField from "../shared/InputField";

const ContactPage = () => {
    const [name, setName] = useState("");
    const [email, setEmail] = useState("");
    const [message, setMessage] = useState("");
    const [inputError, setInputError] = useState("");

    const validateInput = (e) => {
        e.preventDefault();
        if (!name || !email || !message) {
            setInputError("Vennligst fyll inn alle feltene");
            return;
        }
        setInputError("");
        setName("");
        setEmail("");
        setMessage("");
    }

    return (
        <div id="contact-page-container" className="card-container">
            <div id="contact-page-card" className="card">
                <h2>KONTAKT OSS</h2>
                <form onSubmit={validateInput}>
                    {inputError && <InputErrorView error={inputError} />}
                    <div id="input-container">
                        <div id="input-container-left">
                            <InputField
                                label={"Navn"}
                                name={"name"}
                                value={name}
                                onValueChange={setName} />
                            <InputField
                                type={"email"}
                                name={"email"}
                                label={"E-postadresse"}
                                value={email}
                                onValueChange={setEmail} />
                            <button id="contact-submit-btn">SEND</button>
                        </div>
                        <div id="textarea-container">
                            <label>Melding</label>
                            <textarea
                                name="message"
                                placeholder="Skriv en melding til oss her!"
                                value={message} onChange={(e) => setMessage(e.target.value)} />
                        </div>
                    </div>
                </form>
            </div>
        </div>
    )
}

export default ContactPage;