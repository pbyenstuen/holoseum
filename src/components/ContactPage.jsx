import React, { useState } from "react";
import InputField from "./InputField";

const ContactPage = () => {
    const [value1, setValue1] = useState("");
    const [value2, setValue2] = useState("");
    const [value3, setValue3] = useState("");

    return (
        <div id="contact-page-container" className="card-container">
            <div id="contact-page-card" className="card">
                <h2>SKJEMA</h2>
                <form>
                    <div id="input-container">
                        <InputField
                            label={"Label"}
                            value={value1}
                            placeholder={"Skriv her"}
                            onValueChange={setValue1} />
                        <InputField
                            label={"Label"}
                            value={value2}
                            placeholder={"Skriv her"}
                            onValueChange={setValue2} />
                        <InputField
                            label={"Label"}
                            value={value3}
                            placeholder={"Skriv her"}
                            onValueChange={setValue3} />
                    </div>
                    <button id="contact-submit-btn">SUBMIT</button>
                </form>
            </div>
        </div>
    )
}

export default ContactPage;