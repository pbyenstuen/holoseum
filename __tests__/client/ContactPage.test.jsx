import React from "react";
import ReactDOM from "react-dom";
import { act, isDOMComponent, Simulate } from "react-dom/test-utils";
import { MemoryRouter } from "react-router";
import ContactPage from "../../src/client/components/landing-page/ContactPage";

let container;

beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
        writable: true,
        value: jest.fn().mockImplementation(query => ({
            matches: false,
            media: query,
            onchange: null,
            addListener: jest.fn(),
            removeListener: jest.fn(),
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
        }))
    });
});

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    document.body.removeChild(container);
    container = null;
});

const mountComponent = async (component) => {
    await act(async () => {
        await ReactDOM.render(<MemoryRouter>{component}</MemoryRouter>, container);
    });
}

describe("ContactPage", () => {
    it("renders contact form", async () => {
        await mountComponent(<ContactPage />);

        expect(container.innerHTML).toMatchSnapshot();
        expect(container.querySelector("h2").textContent).toEqual("KONTAKT OSS");
        expect(isDOMComponent(container.querySelector("form"))).toEqual(true);
    });

    it("can submit name, email and message", async () => {
        await mountComponent(<ContactPage />);

        Simulate.change(container.querySelector("input[name='name']"), {
            target: { value: "Test" },
        });

        Simulate.change(container.querySelector("input[name='email']"), {
            target: { value: "test@test.no" },
        });

        Simulate.change(container.querySelector("textarea"), {
            target: { value: "Jeg er interessert!" },
        });

        await act(async () => {
            Simulate.submit(container.querySelector("form"));
        });

        expect(container.querySelector("input[name='name']").textContent).toEqual("");
        expect(container.querySelector("input[name='email']").textContent).toEqual("");
        expect(container.querySelector("textarea").textContent).toEqual("");
    });

    it("can show error message for empty fields", async () => {
        await mountComponent(<ContactPage />);

        await act(async () => {
            await Simulate.submit(container.querySelector("form"));
        });

        expect(container.innerHTML).toMatchSnapshot();
        expect(container.querySelector("h4").textContent).toEqual("Vennligst fyll inn alle feltene");
    });
});
