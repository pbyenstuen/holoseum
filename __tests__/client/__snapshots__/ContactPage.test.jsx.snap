// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPage can show error message for empty fields 1`] = `"<div id=\\"contact-page-container\\" class=\\"card-container\\"><section id=\\"contact-page-card\\" class=\\"card\\"><h2>KONTAKT OSS</h2><form><div id=\\"input-error-view\\"><i class=\\"fa fa-exclamation\\" title=\\"Utropstegn\\" aria-label=\\"Utropstegn\\"></i><h4>Vennligst fyll inn alle feltene</h4></div><div id=\\"input-container\\"><div id=\\"input-container-left\\"><label>Navn <input name=\\"name\\" type=\\"text\\" value=\\"\\"></label><label>E-postadresse <input name=\\"email\\" type=\\"email\\" value=\\"\\"></label><button id=\\"contact-submit-btn\\">SEND</button></div><div id=\\"textarea-container\\"><label>Melding</label><textarea name=\\"message\\" placeholder=\\"Skriv en melding til oss her!\\"></textarea></div></div></form></section></div>"`;

exports[`ContactPage renders contact form 1`] = `"<div id=\\"contact-page-container\\" class=\\"card-container\\"><section id=\\"contact-page-card\\" class=\\"card\\"><h2>KONTAKT OSS</h2><form><div id=\\"input-container\\"><div id=\\"input-container-left\\"><label>Navn <input name=\\"name\\" type=\\"text\\" value=\\"\\"></label><label>E-postadresse <input name=\\"email\\" type=\\"email\\" value=\\"\\"></label><button id=\\"contact-submit-btn\\">SEND</button></div><div id=\\"textarea-container\\"><label>Melding</label><textarea name=\\"message\\" placeholder=\\"Skriv en melding til oss her!\\"></textarea></div></div></form></section></div>"`;
